### DISCUSSION TITLE
How do I use the return from Custom Method in the ASYNCUA Server Module

### DISCUSSION THREAD

**Discussion by dibyendumca** [FreeOpcUa/opcua-asyncio#1154] [Q&A]
Hello,
Can anyone provide me the example of How to get the return of a Custom Method defined in the Server Module?
I have defined a method which have a return type as Integer value as below.
```
def StartProcess():
        return 1
```
I have the following code in my Server Module.
```
startProc = await ParamMethods.add_method(spaceNodeId, "StartProcess", CustomFeatures.StartProcess,
                                              [], [ua.NodeId(ua.ObjectIds.UInt32)])
```
I could see the Method from the OPC UA client and can call also.
I want to process some logic as soon as the method is called from a client.
**_My question is, how shall I get the return inside the Server module and use the method return for further use?_**

**Comment by AndreasHeine** [comment]
https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/server-methods.py

https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/server-extension-object-as-method-argument.py#L6

**Reply by dibyendumca** [reply]
**How will be the code in the Server module, to get the return of the Method ?**

**Reply by AndreasHeine** [reply]
> How will be the code in the Server module, to get the return of the Method ?

not sure what you mean...

if you need the result of the call later you need to store it somewhere in memory!
